src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/TargetFrameworkProjectConfigurationDimensionProvider.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Build
+{
+    /// <summary>
+    /// Provides "TargetFramework" project configuration dimension and values.
+    /// </summary>
+    [Export(typeof(IProjectConfigurationDimensionsProvider))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class TargetFrameworkProjectConfigurationDimensionProvider : IProjectConfigurationDimensionsProvider
+    {
+        private const string TargetFrameworkPropertyName = "TargetFramework";
+        private const string DefaultTargetFrameworkValue = "netcoreapp1.0";
+        private readonly IProjectLockService _projectLockService;
+        
+        [ImportingConstructor]
+        public TargetFrameworkProjectConfigurationDimensionProvider(IProjectLockService projectLockService)
+        {
+            Requires.NotNull(projectLockService, nameof(projectLockService));
+
+            _projectLockService = projectLockService;
+        }
+
+        public async Task<IEnumerable<KeyValuePair<string, string>>> GetDefaultValuesForDimensionsAsync(UnconfiguredProject project)
+        {
+            // First target framework is the default one.
+            var targetFrameworks = await GetOrderedTargetFrameworksAsync(project).ConfigureAwait(false);
+            return ImmutableArray.Create(new KeyValuePair<string, string>(TargetFrameworkPropertyName, targetFrameworks.FirstOrDefault()));
```

Comment:
If targetFrameworks.IsEmpty then should this return an empty enumerable as well instead of one item with a null value for targetframework?